components/plugin_marketplace/marketplace_item/marketplace_item.js

```diff
@@ -4,16 +4,208 @@
 
 import {trackEvent} from 'actions/diagnostics_actions.jsx';
 import {localizeMessage} from 'utils/utils';
+import {Constants} from 'utils/constants';
+
+// UpdateVersion renders the version text in the update details, linking out to release notes if available.
+export const UpdateVersion = ({version, releaseNotesUrl}) => {
+    if (!releaseNotesUrl) {
+        return version;
+    }
+
+    return (
+        <a
+            target='_blank'
+            rel='noopener noreferrer'
+            href={releaseNotesUrl}
+        >
+            {version}
+        </a>
+    );
+};
+
+UpdateVersion.propTypes = {
+    version: PropTypes.string.isRequired,
+    releaseNotesUrl: PropTypes.string,
+};
+
+// UpdateDetails renders an inline update prompt for plugins, when available.
+export const UpdateDetails = ({version, releaseNotesUrl, installedVersion, isInstalling, onUpdate}) => {
+    if (!installedVersion || isInstalling) {
+        return null;
+    }
+
+    var isUpdate = false;
+    try {
+        isUpdate = semver.gt(version, installedVersion);
+    } catch (e) {
+        // If we fail to parse the version, assume not an update;
+    }
+
+    if (!isUpdate) {
+        return null;
+    }
+
+    return (
+        <div className={classNames('update')}>
+            <FormattedMessage
+                id='marketplace_modal.list.update_available'
+                defaultMessage='Update available:'
+            />
+            {' '}
+            <UpdateVersion
+                version={version}
+                releaseNotesUrl={releaseNotesUrl}
+            />
+            {' - '}
+            <b>
+                <a onClick={onUpdate}>
+                    <FormattedMessage
+                        id='marketplace_modal.list.update'
+                        defaultMessage='Update'
+                    />
+                </a>
+            </b>
+        </div>
+    );
+};
+
+UpdateDetails.propTypes = {
+    version: PropTypes.string.isRequired,
+    releaseNotesUrl: PropTypes.string,
+    installedVersion: PropTypes.string,
+    isInstalling: PropTypes.bool.isRequired,
+    onUpdate: PropTypes.func.isRequired,
+};
+
+// UpdateConfirmationModal prompts before allowing upgrade, specially handling major version changes.
+export const UpdateConfirmationModal = ({show, name, version, installedVersion, releaseNotesUrl, onUpdate, onCancel}) => {
+    if (!installedVersion) {
+        return null;
+    }
+
+    var isUpdate = false;
+    try {
+        isUpdate = semver.gt(version, installedVersion);
+    } catch (e) {
+        // If we fail to parse the version, assume not an update;
+    }
+
+    if (!isUpdate) {
+        return null;
+    }
+
+    const messages = [(
+        <p key='intro'>
+            <FormattedMessage
+                id='marketplace_modal.list.update_confirmation.message.intro'
+                defaultMessage={`Are you sure you want to update the ${name} plugin to ${version}?`}
```

Comment:
Out of scope of this PR, but I would like to see uns using `v1.2.3` instead of `1.2.3`. It makes it much more clear what this is about.

Reply:
No strong objections, but deferring to avoid having to touch this everywhere else too.